daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/parsers/SequenceChildBases.scala

```diff
@@ -519,8 +519,13 @@ trait EndArrayChecksMixin {
         val minO = erd.minOccurs
         val maxO = erd.maxOccurs
         val isUnbounded = maxO == -1
-        val occurrence = actualOccurs - 1
-
+        val occurrence = {
+          val maybeLastChild = state.infoset.maybeLastChild
```

Comment:
I think it's worth adding a comment explaining why we have this logic for figuring out the number of occurrences of this array, since the obvious logic of something like `pstate.infoset.numChildren` is actually wrong. Maybe something like this?

> at this point, state.infoset is the parent node of the DIArray that we are trying to validate. If any elements were added for this array, then the DIArray will be the last child of that parent node AND it will have the same erd as this parser. If that's not the case, and the parent has no children or it does but the last child has a different erd than what we expect, then it must mean nothing was added for this array and the occurrences are zero.

Reply:
Added

Reply:
I see that the new comment failed the CI's scalafmt check.  Why did the check fail; did scalafmt want to rewrap the comment or what?  I believe there's a way to configure Intellij IDEA to call scalafmt upon save so you can avoid failing the scalafmt check again in the future.  Please bring the [Intellij IDEA Setup](https://cwiki.apache.org/confluence/display/DAFFODIL/IntelliJ+IDEA+Setup) page up to date (what it says about the CLI tests is also out of date) so that developers will know how to avoid failing the check multiple times.